component/azstorage/block_blob.go

```diff
@@ -606,10 +610,35 @@ func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]string, fi *
 	//defer exectime.StatTimeCurrentBlock("WriteFromFile::WriteFromFile")()
 
 	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
-
 	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromFile", name)
+	blockSize := bb.Config.blockSize
+	// if the block size is not set then we configure it based on file size
+	if bb.Config.blockSize == 0 {
+		// get the size of the file
+		stat, err := fi.Stat()
+		if err != nil {
+			log.Err("BlockBlob::WriteFromFile : Failed to get file size %s (%s)", name, err.Error())
+			return err
+		}
+		fileSize := stat.Size()
+		// If bufferSize > (BlockBlobMaxStageBlockBytes * BlockBlobMaxBlocks), then error
+		if fileSize > MaxBlocks {
+			err = errors.New("buffer is too large to upload to a block blob")
+			log.Err("BlockBlob::WriteFromFile : buffer is too large to upload to a block blob %s", name)
+			return err
+		}
+		// If bufferSize <= BlockBlobMaxUploadBlobBytes, then Upload should be used with just 1 I/O request
+		if fileSize <= azblob.BlockBlobMaxUploadBlobBytes {
```

Comment:
Shall we move this block size calculation to a seperate method in utils where you give blocksize and filesize and it returns back the chosen blocksize. That way this code can be reused in future if required and also then you can add UT to try all possible combinations and validate thta we do return correct blocksize.